torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4550,7 +4550,7 @@ def _maybe_evaluate_static(
         # sympy will not simplify the a. The two appeareances of the a will then make value ranges
         # analysis give lose bounds
         new_expr = canonicalize_bool_expr(safe_expand(new_expr))
-        if new_expr.is_number:
+        if len(new_expr.free_symbols) == 0:
```

Comment:
What happened here?

Reply:
If new_expr is a boolean, is_number returns False, when it seems like we should just short circuit

Reply:
(This change shouldn't materially do anything though)